xLights/controllers/ILightThat.cpp

```diff
@@ -0,0 +1,130 @@
+            logger_base.debug("Connected to ILightThat controller model %s.", (const char*)GetFullName().c_str());
+        }
+    } else {
+        _connected = false;
+        logger_base.error("Error connecting to ILightThat controller on %s.", (const char*)_ip.c_str());
+    }
+}
+
+ILightThat::~ILightThat()
+{
+    /*
+    for (const auto& it : _pixelOutputs) {
+        delete it;
+    }
+    _pixelOutputs.clear();*/
+}
+
+#pragma endregion
+
+#pragma region Getters and Setters
+bool ILightThat::SetOutputs(ModelManager* allmodels, OutputManager* outputManager, Controller* controller, wxWindow* parent)
+{
+    static log4cpp::Category& logger_base = log4cpp::Category::getInstance(std::string("log_base"));
+    logger_base.debug("ILightThat Outputs Upload: Uploading to %s", (const char*)_ip.c_str());
+
+    std::string check;
+    UDController cud(controller, outputManager, allmodels, false);
+    auto rules = ControllerCaps::GetControllerConfig(controller);
+    bool success = cud.Check(rules, check);
+
+    if (success) {
+        bool changed = false;
+        wxJSONValue outputConfig = new wxJSONValue();
+        int first_channel = cud.GetFirstOutput()->GetStartChannel();
+        outputConfig["start_address"] = first_channel;
+        //GetOutputConfig(outputConfig);
+        for (int x = 0; x < cud.GetMaxPixelPort(); x++) {
+            UDControllerPort* port = cud.GetControllerPixelPort(x + 1);
+            std::string const proto = port->GetProtocol();
+            if (port->Pixels() > 750) {
+                DisplayError("Max 750 pixels per port for Baldrick Board", parent);
+                return false;
+            }
+            outputConfig["ports"][x]["num_pixels"] = port->Pixels();
+
+            int i = 0;
+            for (auto model : port->GetModels()) {
+                outputConfig["ports"][x]["models"][i]["name"] = model->GetName();
+                int brightness = model->GetBrightness(-1);
+                if (brightness == -1)
+                {
+                    brightness = 100;
+                }
+                
+                outputConfig["ports"][x]["models"][i]["brightness"] = brightness;
+                outputConfig["ports"][x]["models"][i]["start"] = (model->GetStartChannel() - port->GetFirstModel()->GetStartChannel()) / 3;
+                outputConfig["ports"][x]["models"][i]["num_pixels"] = ((model->GetEndChannel() - model->GetStartChannel()) + 1) / 3;
+
+                const std::string colorOrder = model->GetColourOrder("unknown");
+                if (colorOrder != "unknown") {
+                    outputConfig["ports"][x]["models"][i]["colour_order"] = colorOrder;
+                }
+                i++;
+            }
+        }
+        for (int i = cud.GetMaxPixelPort(); i < controller->GetControllerCaps()->GetMaxPixelPort() + 1; i++)
+        {
+            outputConfig["ports"][i]["num_pixels"] = 0;
+        }
+
+        std::string url = "http://" + _ip + "/settings";
```

Comment:
if you plan to support FPP Proxy, this URL wont work. If you use the BaseController::PutURL method it automatedly adds the proxy path for you.

Reply:
That _is_ handy, thanks :)

I'll freely admit I borrowed this code from another driver. If I can remember which one, I'll fix there and PR that too later.